apps/webapp/src/state/swap.test.ts

```diff
@@ -70,4 +70,35 @@ describe('Swap Slice', () => {
     useStore.getState().swap.setAmount('22.44');
     expect(useStore.getState().swap.amount).toBe('22.44');
   });
+
+  test('changing parameters clears simulation', () => {
+    expect(useStore.getState().swap.simulateOutResult).toBeUndefined();
+    useStore.setState(state => {
+      state.swap.simulateOutResult = new ValueView();
+      return state;
+    });
+    expect(useStore.getState().swap.simulateOutResult).toBeDefined();
+
+    // Asset in change resets simulateOutResult
```

Comment:
I would write each of these as a separate `test` so that we know exactly where the point of failure is when a test fails.